### PR TITLE
fix: failing push notifications

### PR DESCRIPTION
**Pull request by Prithpal-Sooriya**
## **Description**

This fixes push notifications so that they are not silent push notifications.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25340?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Create a wallet notification
2. See push notification received

### **Before**

image

### **After**

Screenshot 2024-06-14 at 21 04 55

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/controllers/push-platform-notifications/services/services.ts [hunk 1]
```diff
@@ -195,18 +195,21 @@ export async function listenToPushNotifications(
...
-        if (!notificationData) {
+        // if the payload does not contain data, do nothing
+        if (!data) {
           return;
         }
 
+        const notificationData = {
+          ...data,
+          type: data?.type ?? data?.data?.kind,
+        } as NotificationUnion;
```
Comment (Prithpal-Sooriya) [inline app/scripts/controllers/push-platform-notifications/services/services.ts:211]: God more bad types we need to resolve.

A raw notification DOES NOT have a `type` property, this is something we inject before processing. This adds back this type property if it is missing.

Comment (Prithpal-Sooriya) [reply app/scripts/controllers/push-platform-notifications/services/services.ts:211]: Most of our types are sound. These are edge-cases and artefacts from the portfolio integration.

File: app/scripts/controllers/push-platform-notifications/utils/get-notification-image.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import browser from 'webextension-polyfill';
+
+export async function getNotificationImage() {
+  const iconUrl = await browser.runtime.getURL('../../images/icon-64.png');
```
Comment (Prithpal-Sooriya) [inline app/scripts/controllers/push-platform-notifications/utils/get-notification-image.ts:4]: This correctly gets the browser image (used on other notifications - e.g. transaction notifications)

**Review by matteoscurati — APPROVED** [review]
Well done!

**Review by dovydas55 — APPROVED** [review]

**Review by chakra-guy — APPROVED** [review]
LGTM

**Pull request merged**